tests/roles/run_tests/templates/cluster-template-cluster.yaml

```diff
@@ -27,7 +27,7 @@ spec:
   controlPlaneEndpoint:
     host: ${ CLUSTER_APIENDPOINT_HOST }
     port: ${ CLUSTER_APIENDPOINT_PORT }
-  noCloudProvider: true
+  cloudProviderEnabled: false
```

Comment:
This however does not change the logic. The error message is about conflicting values. This should not happen when only one value is provided right?

Reply:
yes, that is what I am testing, not sure why its conflicting.  I am in the VM where the test is running to check what are the final values of noCloudProvider and cloudProviderEnabled when we apply it. Lets see

Reply:
setting only cloudProviderEnabled works fine. I will run periodic ci test again, I need to check what is happening in the failure case.

Reply:
in the generated yaml file that is being applied I only see noCloudProvider:
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
```
kind: Metal3Cluster
metadata:
  name: test1
  namespace: metal3
spec:
  controlPlaneEndpoint:
    host: 192.168.111.249
    port: 6443
  noCloudProvider: true
```
I have no clue why it says there is a conflict; something is wrong. I am running tests locally to debug more.

Reply:
I have figured it out, it is because we are setting // +kubebuilder:default=true for cloudProviderEnabled, in validation webhook it fails because it gets default value of cloudProviderEnabled instead of nil. 
We can fix this by adding defaulting webhook, and removing // +kubebuilder:default=true , I am working on it. 
One weird thing is this does not happen in tests here : https://github.com/metal3-io/cluster-api-provider-metal3/blob/7aaf7b896c7939776198e9a3d19f2ec5a963c2b0/api/v1beta1/metal3cluster_webhook_test.go#L121 ,